### DISCUSSION TITLE
Mirth service is not getting restarted

### DISCUSSION THREAD

**Discussion by Shahabuddin222** [nextgenhealthcare/connect#6175] [Q&A]
Mirth service is not getting restarted if we use Recovery option which is present under Windows Services under Recovery tab.
There is any alternative thing to restart if the service is getting stopped for some issue
![Screenshot 2024-04-16 145607](https://github.com/nextgenhealthcare/connect/assets/167187884/605dc960-e5ec-4463-8239-9a64e78b96e1)

**Comment by Shahabuddin222** [comment]
Anyone here who can help me out

**Comment by pacmano1** [comment]
Why is the service getting stopped to begin with?

What does the mirth log file say when it stops unexpectedly?

Mirth almost never stops on its own, it is extremely reliable.

In fact it is so reliable, I've never had to rely on the operating system to restart it.

**Reply by Shahabuddin222** [reply]
So we are using Databases from the different server and when we restart SQL services sometimes Mirth service will Stop sometimes it keep running and when its stopped because of some error we are not able to use the OS feature to restart the Mirth service again

**Comment by ab-mg-23** [comment]
[Restart Actions](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-service_failure_actions_flag) only happen: 

> If this member is TRUE and the service has configured failure actions, the failure actions are queued if the service process terminates without reporting a status of SERVICE_STOPPED or if it enters the SERVICE_STOPPED state but the dwWin32ExitCode member of the [SERVICE_STATUS](https://learn.microsoft.com/en-us/windows/desktop/api/winsvc/ns-winsvc-service_status) structure is not ERROR_SUCCESS (0).

If something on the system is using the WinService API or sc.exe to stop the Mirth Service, that's WAD. If you need to find out what/whom is making the service stop that way, you can configure Windows Auditing on [Audit Handle Manipulation and Audit Other Object Access Events - Success](https://learn.microsoft.com/en-us/archive/msdn-technet-forums/aeeecc07-368f-4f68-b773-6af70eec2995). However, doing so will generate a large number of entries so you should only enable it for as long as you need to find the culprit.

**Reply by Shahabuddin222** [reply]
Thanks for the help i will try to find the issue why its getting stopped. 
But still my question will be same that can we restart Mirth service in automatic way

**Comment by pacmano1** [comment]
Moving to discussions until this is a confirmed issue.

**Comment by rjkroll** [comment]
You could something like this Pshell script and schedule it as a Windows Task 



```
$ServiceName = "mcservice.exe" # whatever service you want to monitor

while ($true) {
    $Service = Get-Service -Name $ServiceName
    if ($Service.Status -ne "Running") {
        Write-Host "$($Service.DisplayName) service is not running. Restarting..." -ForegroundColor Yellow
        Start-Service -Name $ServiceName
        Start-Sleep -Seconds 10 # Wait for the service to start
        $Service.Refresh() # Refresh the service object
        if ($Service.Status -eq "Running") {
            Write-Host "$($Service.DisplayName) service has been restarted successfully." -ForegroundColor Green
        } else {
            Write-Host "$($Service.DisplayName) service failed to restart." -ForegroundColor Red
        }
    }
    Start-Sleep -Seconds 60 # Check the service status every 60 seconds
```

**Reply by pacmano1** [reply]
Nice (really).    

I think "someone is restarting SQL server and not coordinating with the integration team" is vary poor IT management.

**Reply by Shahabuddin222** [reply]
Yes we also have PowerShell script but we want to use as a secondary option not as a primary.
